spec/models/mixins/supports_feature_mixin_spec.rb

```diff
@@ -338,28 +332,24 @@ def initialize(values = {})
   private
 
   def define_model(class_name, parent, supporting_values = {})
-    define_supporting_class(class_name, parent, supporting_values) do |r|
-      r.table_name = "vms" if parent.ancestors.include?(ActiveRecord::Base)
-      yield(r) if block_given?
-    end
+    define_supporting_class(class_name, parent, supporting_values)
   end
 
   def define_subclass(module_name, parent, supports_values = {})
     define_supporting_class("#{module_name}::#{parent.name}", parent, supports_values)
   end
 
-  # these descendants are stored in a cache in active support
-  # this cleans out those values so future runs do not have bogus classes
-  # this causes sporadic test failures.
-  def cleanup_subclass(parent, children)
-    tracker = ActiveSupport::DescendantsTracker.subclasses(parent)
-    tracker&.reject! { |child| children.include?(child) }
-  end
-
   def define_supporting_class(class_name, parent, supports_values = {})
     child = Class.new(parent) do
       include SupportsFeatureMixin unless parent.respond_to?(:supports?)
 
+      self.table_name = "vms" if parent.ancestors.include?(ActiveRecord::Base)
+      if parent == ExtManagementSystem
+        def self.default_blacklisted_event_names
```

Comment:
Can you add a comment about the HACK here and why it's needed - a casual reader wil just not understand this.

I am concerned that default_blacklisted_event_names isn't the only place this issue is going to creep up, but this is probably better than nothing for now.